File: backup/mazury-holiday_backup_2026-05-25/fix_stranda.py
import re

with open('src/data/stranda-apartments.ts', 'r') as f:
    content = f.read()

# The file has imports at the top, then export const STRANDA_APARTMENTS = { ... };
header_match = re.match(r'^([\s\S]*?export const STRANDA_APARTMENTS: Record<string, ApartmentTemplateData> = \{\n)([\s\S]*?)(\n\};\n?)$', content)

if not header_match:
    print("Could not match file structure")
    exit(1)

header = header_match.group(1)
body = header_match.group(2)
footer = header_match.group(3)

# Split body by blocks
blocks = re.split(r'\n    \'([A-Za-z0-9_ -]+)\': \{', "\n" + body)

apartments = {}
for i in range(1, len(blocks), 2):
    key = blocks[i]
    val = "{" + blocks[i+1].rstrip()
    if val.endswith(","):
        val = val[:-1]
    
    # Exclude the bad keys
    if key in ["C_Studio", "C_1_Sypialnia", "C_2_Sypialnie", "C Studio", "C 1-Sypialnia", "C 2-Sypialnie"]:
        continue
    
    # We keep the last one to naturally deduplicate
    apartments[key] = val

def replace_gallery(block, ido_id):
    import os
    img_dir = f"/Users/karolbohdanowicz/my-ai-agents/mazury-holiday/public/images/stranda/"
    files = []
    try:
        all_files = os.listdir(img_dir)
        files = [f for f in all_files if f.startswith(f"ido_{ido_id}_") and f.endswith(".jpg")]
        files.sort(key=lambda x: int(x.split('_')[2].split('.')[0]))
    except Exception as e:
        print(f"Error reading dir for {ido_id}: {e}")
        pass
        
    if not files:
        return block
        
    hero = f'getAssetPath("/images/stranda/{files[0]}")'
    images_arr = ",\n                ".join([f'getAssetPath("/images/stranda/{f}")' for f in files])
    
    new_gallery = f'''gallery: {{
            "heroImage": {hero},
            "images": [
                {images_arr}
            ]
        }}'''
    
    block = re.sub(r'gallery:\s*\{[\s\S]*?\}', new_gallery, block)
    return block

def replace_title(block, title):
    return re.sub(r"title: `[^`]+`", f"title: `{title}`", block)

# Fix specific apartments
for key in ['B304', 'B305', 'C_1BEDROOM']:
    if key in apartments:
        apartments[key] = replace_gallery(apartments[key], '44')

if 'C_STUDIO' in apartments:
    apartments['C_STUDIO'] = replace_gallery(apartments['C_STUDIO'], '32')
    apartments['C_STUDIO'] = replace_title(apartments['C_STUDIO'], 'C Studio')

if 'C_1BEDROOM' in apartments:
    apartments['C_1BEDROOM'] = replace_title(apartments['C_1BEDROOM'], 'C z jedną sypialnią')

if 'C_2BEDROOM' in apartments:
    apartments['C_2BEDROOM'] = replace_gallery(apartments['C_2BEDROOM'], '43')
    apartments['C_2BEDROOM'] = replace_title(apartments['C_2BEDROOM'], 'C z dwoma sypialniami')

# Fix C304, C301, B402, C404, A302 which might have C_Generic
fixes = {'C304': '30', 'C301': '40', 'B402': '24', 'C404': '41', 'A302': '42'}
for k, ido in fixes.items():
    if k in apartments:
        apartments[k] = replace_gallery(apartments[k], ido)

# Reassemble
new_body = ""
for k, v in apartments.items():
    new_body += f"    '{k}': {v},\n"

# Remove the trailing comma from the last item
new_body = new_body.rstrip().rstrip(",") + "\n"

with open('src/data/stranda-apartments.ts', 'w') as f:
    f.write(header + new_body + footer)

print("Fixed stranda-apartments.ts")

File: czarter-houseboat/fix_stranda.js
const fs = require('fs');
const content = fs.readFileSync('src/data/stranda-apartments.ts', 'utf8');

const startIndex = content.indexOf('export const strandaApartments: Record<string, Apartment> = {');
if (startIndex === -1) {
    console.log("Could not find start");
    process.exit(1);
}
const startOfObj = startIndex + 'export const strandaApartments: Record<string, Apartment> = {'.length;
const endOfObj = content.lastIndexOf('};');

const header = content.substring(0, startOfObj) + '\n';
const body = content.substring(startOfObj, endOfObj);
const footer = '\n' + content.substring(endOfObj);

const blocks = body.split(/\n    '([A-Za-z0-9_ -]+)': \{/);
const apartments = {};

const badKeys = [
    "C_Studio", "C_1_Sypialnia", "C_2_Sypialnie", 
    "C Studio", "C 1-Sypialnia", "C 2-Sypialnie", "C 2-Sypialni"
];

for (let i = 1; i < blocks.length; i += 2) {
    const key = blocks[i];
    let val = "{" + blocks[i+1].trimEnd();
    if (val.endsWith(",")) {
        val = val.slice(0, -1);
    }
    
    if (badKeys.includes(key)) continue;
    
    // Naturally deduplicate by keeping the last encountered version of the key
    apartments[key] = val;
}

function replaceGallery(block, idoId) {
    const imgDir = '/Users/karolbohdanowicz/my-ai-agents/mazury-holiday/public/images/stranda/';
    let files = [];
    try {
        const allFiles = fs.readdirSync(imgDir);
        files = allFiles.filter(f => f.startsWith(`ido_${idoId}_`) && f.endsWith('.jpg'));
        files.sort((a, b) => parseInt(a.split('_')[2]) - parseInt(b.split('_')[2]));
    } catch (e) {
        console.log(`Error reading dir for ${idoId}: ${e.message}`);
    }
    
    if (files.length === 0) {
        console.log(`No images found for idoId ${idoId}`);
        return block;
    }
    
    const hero = `getAssetPath("/images/stranda/${files[0]}")`;
    const imagesArr = files.map(f => `getAssetPath("/images/stranda/${f}")`).join(',\n                ');
    
    const newGallery = `gallery: {
            heroImage: ${hero},
            images: [
                ${imagesArr}
            ]
        }`;
        
    return block.replace(/gallery:\s*\{[\s\S]*?\}/, newGallery);
}

function replaceTitle(block, title) {
    let replaced = block.replace(/title:\s*`[^`]+`/, `title: \`${title}\``);
    replaced = replaced.replace(/title:\s*'[^']+'/, `title: \`${title}\``);
    return replaced;
}

// Map IDO to apartments that have C_Generic
const mappings = {
    'B304': '44', 'B305': '44',
    'C304': '30', 'C301': '40', 
    'B402': '24', 'C404': '41', 
    'A302': '42', 'C402': '46',
    'C403': '41', 'C_1BEDROOM': '44',
    'C_2BEDROOM': '43', 'C_STUDIO': '32'
};

for (const [key, ido] of Object.entries(mappings)) {
    if (apartments[key]) {
        apartments[key] = replaceGallery(apartments[key], ido);
    }
}

if (apartments['C_1BEDROOM']) apartments['C_1BEDROOM'] = replaceTitle(apartments['C_1BEDROOM'], 'C z jedną sypialnią');
if (apartments['C_2BEDROOM']) apartments['C_2BEDROOM'] = replaceTitle(apartments['C_2BEDROOM'], 'C z dwoma sypialniami');
if (apartments['C_STUDIO']) apartments['C_STUDIO'] = replaceTitle(apartments['C_STUDIO'], 'C Studio');

let newBody = "";
const keys = Object.keys(apartments);
for (let i = 0; i < keys.length; i++) {
    const k = keys[i];
    newBody += `    '${k}': ${apartments[k]}${i === keys.length - 1 ? '' : ',\n\n'}`;
}

fs.writeFileSync('src/data/stranda-apartments.ts', header + newBody + footer);
console.log("Fixed stranda-apartments.ts");
